src/bluecherry.c

```diff
@@ -439,6 +445,85 @@ static int _bluecherry_dtls_recv(void *ctx, unsigned char *buf, size_t len)
   return recv(sock, buf, len, 0);
 }
 
+static esp_err_t bluecherry_connect(void)
+{
+    int ret;
+    const int handshake_total_timeout_s = 10; /* tune as needed */
```

Comment:
This is a magic value, this should not be in the code. It can be in the header file or as a constant on top. 10 seconds seems small for slower NB-IoT connections. Would 60 seconds not be better?